src/cluster_legacy.h

```diff
@@ -277,6 +279,25 @@ static_assert(offsetof(clusterMsg, data) == 2256, "unexpected field offset");
                                               primary is up. */
 #define CLUSTERMSG_FLAG0_EXT_DATA (1 << 2) /* Message contains extension data */
 
+typedef struct {
+    char sig[4];                  /* Signature "RCmb" (Cluster message bus). */
+    uint32_t totlen;              /* Total length of this message */
+    uint16_t ver;                 /* Protocol version, currently set to CLUSTER_PROTO_VER. */
+    uint16_t port;                /* Primary port number (TCP or TLS). */
+    uint16_t type;                /* Message type */
+    union clusterMsgData data;
+} clusterMsgPubsub;
+
+
+static_assert(offsetof(clusterMsgPubsub, sig) == 0, "unexpected field offset");
+static_assert(offsetof(clusterMsgPubsub, totlen) == 4, "unexpected field offset");
+static_assert(offsetof(clusterMsgPubsub, ver) == 8, "unexpected field offset");
+static_assert(offsetof(clusterMsgPubsub, port) == 10, "unexpected field offset");
+static_assert(offsetof(clusterMsgPubsub, type) == 12, "unexpected field offset");
+static_assert(offsetof(clusterMsgPubsub, data) == 16, "unexpected field offset");
```

Comment:
For these, it's important that they are at the same offset as in the clusterMsg struct, so we can let the assert can check that instead of the exact numbers. We already have an assert for the exact offsets in the clusterMsg struct, so it's actually equivalent, but I think it's more readable and show the purpose to the reader if we compare with clusterMsg.

```C
static_assert(offsetof(clusterMsgPubsub, totlen) == offsetof(clusterMsg, totlen),
              "unexpected field offset");
```